extras/src/test/java/com/google/gson/typeadapters/RuntimeTypeAdapterFactoryTest.java

```diff
@@ -167,6 +167,25 @@ public void testSerializeCollidingTypeFieldName() {
     }
   }
 
+  public void testSerializeWrappedNullValue() {
+    TypeAdapterFactory billingAdapter = RuntimeTypeAdapterFactory.of(BillingInstrument.class)
```

Comment:
I can't see how: this  IS a bug in `RuntimeTypeAdapterFactory`. Not in the GSON core.

As far as I understand, the `extras` package (where this class is located) is intended as a set of additional helper classes available for people to use, but it is distributed as source, I assume driven out of a desire to keep the GSON core light-weight. As far as I know, the compiled class is not packaged and distributed. I, as many others on the web have done, manually include and build the source ourselves in our own projects to make use of it.

(This is quite distinct from the `internal` package which includes source that is compiled, shipped and used to support the GSON core public API. However, it should not be called directly.)

So, in my use of `RuntimeTypeAdapterFactory` I have found the bug, fixed it and wish to share it with other people who use the class as well. Otherwise, what is this class for?